biogeochem/FatesLandUseChangeMod.F90

```diff
@@ -0,0 +1,307 @@
+  public :: get_luh_statedata
+
+
+  ! module data
+  integer, parameter :: max_luh2_types_per_fates_lu_type = 5
+
+  ! Define the mapping from the luh2 state names to the aggregated fates land use categories
+  type :: luh2_fates_lutype_map
+
+     character(len=5), dimension(12) :: state_names = &
+                   [character(len=5) ::  'primf','primn','secdf','secdn', &
+                                         'pastr','range', 'urban', &
+                                         'c3ann','c4ann','c3per','c4per','c3nfx']
+     integer, dimension(12) :: landuse_categories = &
+                                [primaryland, primaryland, secondaryland, secondaryland, &
+                                 pastureland, rangeland, fates_unset_int, &
+                                 cropland, cropland, cropland, cropland, cropland]
+
+     contains
+
+       procedure :: GetIndex => GetLUCategoryFromStateName
+
+  end type luh2_fates_lutype_map
+
+
+  ! 03/10/2023 Created By Charlie Koven
+  ! ============================================================================
+
+contains
+
+  ! ============================================================================
+  subroutine get_landuse_transition_rates(bc_in, landuse_transition_matrix)
+
+
+    ! The purpose of this routine is to ingest the land use transition rate information that the host model has read in from a dataset,
+    ! aggregate land use types to those being used in the simulation, and output a transition matrix that can be used to drive patch
+    ! disturbance rates.
+
+    ! !ARGUMENTS:
+    type(bc_in_type) , intent(in) :: bc_in
+    real(r8), intent(inout) :: landuse_transition_matrix(n_landuse_cats, n_landuse_cats)  ! [m2/m2/day]
+
+    ! !LOCAL VARIABLES:
+    type(luh2_fates_lutype_map) :: lumap
+    integer :: i_donor, i_receiver, i_luh2_transitions, i_luh2_states
+    character(5) :: donor_name, receiver_name
+    character(14) :: transition_name
+    real(r8) :: urban_fraction
+    real(r8) :: temp_vector(hlm_num_luh2_transitions)
+    logical  :: modified_flag
+
+    ! zero the transition matrix and the urban fraction
+    landuse_transition_matrix(:,:) = 0._r8
+    urban_fraction = 0._r8
+
+    use_luh_if: if ( hlm_use_luh .eq. itrue ) then
+
+       ! Check the LUH data incoming to see if any of the transitions are NaN
+       temp_vector = bc_in%hlm_luh_transitions
+       call CheckLUHData(temp_vector,modified_flag)
+       if (.not. modified_flag) then
+          ! identify urban fraction so that it can be factored into the land use state output
+          urban_fraction = bc_in%hlm_luh_states(findloc(bc_in%hlm_luh_state_names,'urban',dim=1))
+       end if
+
+       !!TODO: may need some logic here to ask whether or not ot perform land use change on this timestep. current code occurs every day.
+       !!If not doing transition every day, need to update units.
+
+       transitions_loop: do i_luh2_transitions = 1, hlm_num_luh2_transitions
```

Comment:
is hlm_num_luh2_transitions set by the user, or is it directly interpreted by getting the dimensions from the input transition file?  It seems like it would be more robust to diagnose this value when needed instead of passing it in.